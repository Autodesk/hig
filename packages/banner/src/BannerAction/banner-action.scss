@import "../tokens.scss";

.hig__banner__action {
  display: flex;
  align-items: center;
  padding: $banner-action-padding-y $banner-action-spacing-x $banner-action-padding-y 0;
  border-right-width: 1px;
  border-right-style: solid;
  margin-left: $banner-action-spacing-x;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    .hig__banner--wrap-content & {
      border-right: 0;
    }
  }

  .hig__banner__interactions--wrap-actions & {
    padding-right: 0;
    border-right: 0;
    margin-left: 0;
  }
}

.hig--light-theme {
  .hig__banner--urgent &.hig__banner__action {
    border-right-color: map-get($banner-hig-light-theme, border-color--urgent);
  }

  .hig__banner--warning &.hig__banner__action {
    border-right-color: map-get($banner-hig-light-theme, border-color--warning);
  }

  .hig__banner--primary &.hig__banner__action {
    border-right-color: map-get($banner-hig-light-theme, border-color--primary);
  }

  .hig__banner--complete &.hig__banner__action {
    border-right-color: map-get($banner-hig-light-theme, border-color--complete);
  }
}

.hig--matrix-theme {
  .hig__banner--urgent &.hig__banner__action {
    border-right-color: map-get($banner-matrix-theme, border-color--urgent);
  }

  .hig__banner--warning &.hig__banner__action {
    border-right-color: map-get($banner-matrix-theme, border-color--warning);
  }

  .hig__banner--primary &.hig__banner__action {
    border-right-color: map-get($banner-matrix-theme, border-color--primary);
  }

  .hig__banner--complete &.hig__banner__action {
    border-right-color: map-get($banner-matrix-theme, border-color--complete);
  }
}