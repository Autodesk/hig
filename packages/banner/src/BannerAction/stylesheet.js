import constants from "../constants";

export default function stylesheet(props, themeData) {
  const vars = constants(themeData);
  const { stylesheet: customStylesheet } = props;
  const styles = {
    display: "flex",
    alignItems: "center",
    padding: `${vars.bannerActionPaddingY} 0`,
    marginLeft: vars.bannerActionSpacingX,

    "&:first-of-type": {
      marginLeft: 0,
    },

    ...(props.isWrappedContent && {
      "&:last-of-type": {
        borderRight: 0,
      },
    }),
  };
  return customStylesheet ? customStylesheet(styles, props, themeData) : styles;
}
