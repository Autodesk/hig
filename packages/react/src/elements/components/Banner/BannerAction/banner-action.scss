@import "../tokens.scss";

.hig__banner__action {
  display: flex;
  align-items: center;
  padding: $banner-action-padding-y $banner-action-spacing-x $banner-action-padding-y 0;
  border-right-width: 1px;
  border-right-style: solid;
  margin-left: $banner-action-spacing-x;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    .hig__banner--wrap-content & {
      border-right: 0;
    }
  }

  .hig__banner--urgent & {
    border-right-color: $banner-border-color--urgent;
  }

  .hig__banner--warning & {
    border-right-color: $banner-border-color--warning;
  }

  .hig__banner--primary & {
    border-right-color: $banner-border-color--primary;
  }

  .hig__banner--complete & {
    border-right-color: $banner-border-color--complete;
  }

  .hig__banner__interactions--wrap-actions & {
    padding-right: 0;
    border-right: 0;
    margin-left: 0;
  }
}
