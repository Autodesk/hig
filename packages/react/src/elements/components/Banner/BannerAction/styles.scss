.hig__banner__action {
  @import "../variables.scss";

  display: flex;
  align-items: center;
  padding: $action-padding-y $action-spacing-x $action-padding-y 0;
  border-right-width: 1px;
  border-right-style: solid;
  margin-left: $action-spacing-x;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    .hig__banner--wrap-content & {
      border-right: 0;
    }
  }

  .hig__banner--error & {
    border-right-color: $border-color--error;
  }

  .hig__banner--warning & {
    border-right-color: $border-color--warning;
  }

  .hig__banner--primary & {
    border-right-color: $border-color--primary;
  }

  .hig__banner--success & {
    border-right-color: $border-color--success;
  }

  .hig__banner__interactions--wrap-actions & {
    padding-right: 0;
    border-right: 0;
    margin-left: 0;
  }
}
