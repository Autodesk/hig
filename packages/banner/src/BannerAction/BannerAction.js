import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import ThemeContext from "@hig/theme-context";
import stylesheet from "./stylesheet";

/**
 * @typedef {Object} BannerActionProps
 * @property {any} children
 * @property {Function} stylesheet
 */

/**
 * @param {BannerActionProps} props
 * @returns {JSX.Element}
 */
export default function BannerAction({
  children,
  stylesheet: customStylesheet
}) {
  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => (
        <div
          className={css(
            stylesheet({ stylesheet: customStylesheet }, resolvedRoles)
          )}
        >
          {children}
        </div>
      )}
    </ThemeContext.Consumer>
  );
}

BannerAction.propTypes = {
  children: PropTypes.node,
  stylesheet: PropTypes.func
};
