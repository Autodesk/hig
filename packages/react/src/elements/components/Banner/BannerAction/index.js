import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

/**
 * @typedef {Object} BannerActionProps
 * @property {any} children
 */

/**
 * @param {BannerActionProps} props
 * @returns {JSX.Element}
 */
export default function BannerAction({ children }) {
  return <div className="hig__banner__action">{children}</div>;
}

BannerAction.propTypes = {
  children: PropTypes.node
};
