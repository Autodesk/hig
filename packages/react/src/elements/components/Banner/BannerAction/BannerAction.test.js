import React from "react";
import renderer from "react-test-renderer";

import BannerAction from "./BannerAction";

describe("banner/BannerAction/BannerAction", () => {
  it("renders with children", () => {
    const action = <BannerAction>Hello</BannerAction>;
    const tree = renderer.create(action).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
