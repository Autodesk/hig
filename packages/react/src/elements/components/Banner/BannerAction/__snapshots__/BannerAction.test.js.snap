// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner/BannerAction/BannerAction renders with children 1`] = `
<div
  className="hig__banner__action"
>
  Hello
</div>
`;
